### PR TITLE
move stdlib: type_info::type_name()

### PR DESCRIPTION
**Pull request by msmouse**
### Description

Prepare for emitting

```
struct CreateTableEvent {
    handle: u128,
    key_type: String,
    value_type: String,
}
```

### Test Plan
unittest

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2122)

### CODE REVIEW

File: aptos-move/framework/tests/move_unit_test.rs [hunk 1]
```diff
@@ -41,10 +41,15 @@ pub fn aptos_test_natives() -> NativeFunctionTable {
 }
 
 #[test]
-fn move_unit_tests() {
```
Comment (davidiw) [inline aptos-move/framework/tests/move_unit_test.rs:44]: wtf, your file name?

File: aptos-move/framework/aptos-stdlib/sources/type_info.move [hunk 2]
```diff
@@ -18,6 +20,7 @@ module aptos_std::type_info {
     }
 
     public native fun type_of<T>(): TypeInfo;
+    public native fun full_name<T>(): string::String;
```
Comment (davidiw) [inline aptos-move/framework/aptos-stdlib/sources/type_info.move:23]: maybe full_path? or just path?

**Review by davidiw — APPROVED** [review]
thanks batman.

File: aptos-move/framework/aptos-stdlib/sources/type_info.move (hunk 2 above)
Comment (msmouse) [reply aptos-move/framework/aptos-stdlib/sources/type_info.move:23]: @wrwg opinion?

Comment (wrwg) [reply aptos-move/framework/aptos-stdlib/sources/type_info.move:23]: Hmm... looks asymmetric. `type_info::name_of<T>()`?

Comment (msmouse) [reply aptos-move/framework/aptos-stdlib/sources/type_info.move:23]: works for me and better than "path". originally I named it `type_info::type_name()`

**Pull request merged**